core/staking/accounting_snapshot.go

```diff
@@ -34,6 +34,9 @@ func (a *Accounting) serialiseStakingAccounts() ([]byte, error) {
 	accounts := make([]*types.StakingAccount, 0, len(a.hashableAccounts))
 	a.log.Debug("serialsing staking accounts", logging.Int("n", len(a.hashableAccounts)))
 	for _, acc := range a.hashableAccounts {
+		// dedup transactions with the same eth hash from different block heights and recalc the balance
```

Comment:
Whats the reasoning behind doing it at both serialisation and restoring?

Reply:
restoring is for fixing the existing in memory situation on the coming protocol upgrade. serialisation is for it to never be serialised in a bad state again.